stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+        The location of the maximum.
+    sig1:
+        The width parameter for the exponential.
+
+    Returns
+    -------
+    The y values for exponential flare.
+    """
+    A = jnp.atleast_1d(mean_params["A"])[:, jnp.newaxis]
+    t0 = jnp.atleast_1d(mean_params["t0"])[:, jnp.newaxis]
+    sig = jnp.atleast_1d(mean_params["sig"])[:, jnp.newaxis]
+
+    return jnp.sum(A * jnp.exp(-jnp.abs(t - t0) / (2 * (sig**2))), axis=0)
+
+
+def _constant(t, mean_params):
+    """A constant mean shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Constant amplitude of the flare.
+
+    Returns
+    -------
+    The constant value.
+    """
+    return mean_params["A"] * jnp.ones_like(t)
+
+
+def _skew_gaussian(t, mean_params):
+    """A skew gaussian flare shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Amplitude of the flare.
+    t0:
+        The location of the maximum.
+    sig1:
+        The width parameter for the rising edge.
+    sig2:
+        The width parameter for the falling edge.
+
+    Returns
+    -------
+    The y values for skew gaussian flare.
+    """
+    A = jnp.atleast_1d(mean_params["A"])[:, jnp.newaxis]
+    t0 = jnp.atleast_1d(mean_params["t0"])[:, jnp.newaxis]
+    sig1 = jnp.atleast_1d(mean_params["sig1"])[:, jnp.newaxis]
+    sig2 = jnp.atleast_1d(mean_params["sig2"])[:, jnp.newaxis]
+
+    return jnp.sum(
+        A
+        * jnp.where(
+            t > t0,
+            jnp.exp(-((t - t0) ** 2) / (2 * (sig2**2))),
+            jnp.exp(-((t - t0) ** 2) / (2 * (sig1**2))),
+        ),
+        axis=0,
+    )
+
+
+def _skew_exponential(t, mean_params):
+    """A skew exponential flare shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Amplitude of the flare.
+    t0:
+        The location of the maximum.
+    sig1:
+        The width parameter for the rising edge.
+    sig2:
+        The width parameter for the falling edge.
+
+    Returns
+    -------
+    The y values for exponential flare.
+    """
+    A = jnp.atleast_1d(mean_params["A"])[:, jnp.newaxis]
+    t0 = jnp.atleast_1d(mean_params["t0"])[:, jnp.newaxis]
+    sig1 = jnp.atleast_1d(mean_params["sig1"])[:, jnp.newaxis]
+    sig2 = jnp.atleast_1d(mean_params["sig2"])[:, jnp.newaxis]
+
+    return jnp.sum(
+        A
+        * jnp.where(
+            t > t0,
+            jnp.exp(-(t - t0) / (2 * (sig2**2))),
+            jnp.exp((t - t0) / (2 * (sig1**2))),
+        ),
+        axis=0,
+    )
+
+
+def _fred(t, mean_params):
+    """A fast rise exponential decay (FRED) flare shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Amplitude of the flare.
+    t0:
+        The location of the maximum.
+    phi:
+        Symmetry parameter of the flare.
+    delta:
+        Offset parameter of the flare.
```

Comment:
What's the difference between `t0` and the offset parameter `delta` here?

Reply:
`t0` is the maxima of the flare shape, and `delta` is an offset parameter which shifts the whole flare by the value delta. (like a horizontal transform)